checkbox-support/checkbox_support/scripts/run_watcher.py

```diff
@@ -93,124 +90,171 @@ def run(self):
                 continue
-            self._callback([e["MESSAGE"] for e in j if e and "MESSAGE" in e])
+            self._process_lines(
+                [e["MESSAGE"] for e in j if e and "MESSAGE" in e]
+            )
 
-    def _callback(self, lines):
+    def _process_lines(self, lines):
         for line in lines:
             line_str = str(line)
             logger.debug(line_str)
-            self._storage_strategy.do_callback(line_str)
+            self.callback(line_str)
 
-    def _no_storage_timeout(self, signum, frame):
+    def _storage_info_helper(self, reserve, mounted_partition=""):
         """
-        define timeout feature.
-
-        timeout and return failure if there is no usb insertion/removal
-        detected after ACTION_TIMEOUT secs
+        Reserve or removal the detected storage info.
+
+        write the info we got in this script to $PLAINBOX_SESSION_SHARE
+        so the other jobs, e.g. read/write test, could know more information,
+        for example the partition it want to try to mount.
+
+        :param reserve:
+            type: Boolean
+            - True: backup the info of storage partition to PLAINBOX_SESSION_SHARE
+            - False: remove the backup file from PLAINBOX_SESSION_SHARE
+        :param storage_type:
+            type: String
+            - Type of storage. e.g. usb, mediacard and thunderbolt
+        :param mounted_partition:
+            type: String
+            - The name of partition. e.g. sda1, nvme1n1p1 etc...
         """
-        logger.error(
-            "no {} storage {} was reported in systemd journal".format(
-                self.args.storage_type, self.args.testcase
+        plainbox_session_share = os.environ.get("PLAINBOX_SESSION_SHARE")
+        if not plainbox_session_share:
+            logger.error("no env var PLAINBOX_SESSION_SHARE")
+            sys.exit(1)
+
+        # TODO: Should name the file by the value of storage_type variable as
+        #       prefix. e.g. thunderbolt_insert_info, mediacard_insert_info.
+        #       Since usb_insert_info is used by usb_read_write script, we
+        #       should refactor usb_read_write script to adopt different files
+        file_name = "usb_insert_info"
+
+        # backup the storage partition info
+        if reserve and mounted_partition:
+            logger.info(
+                "cache file {} is at: {}".format(
+                    file_name, plainbox_session_share
+                )
             )
-        )
-        sys.exit(1)
+            file_to_share = open(
+                os.path.join(plainbox_session_share, file_name), "w"
+            )
+            file_to_share.write(mounted_partition + "\n")
+            file_to_share.close()
+
+        # remove the back info
+        if not reserve:
+            file_to_share = os.path.join(plainbox_session_share, file_name)
+            with contextlib.suppress(FileNotFoundError):
+                os.remove(file_to_share)
```

Comment:
This is troubles waiting to happen, how did this job run to here without failing if the file was never actually created? I assume this would happen only if you crash while trying to insert but the dependency would not let us run right?

Reply:
Maybe I understood why this happens, if you have more than 1 job then only the first "remove" job is going to work. This is debeatably wrong though, as if if we have an ordering like the following, the process if broken: 
1. insert a
2. use info a
3. insert b
4. remove a
5. use info b <- BOOM

> Note: I don't know if this is what is going on here, just a theory, correct me if I'm wrong

Reply:
I have created an jira ticket to follow up on this issue https://warthogs.atlassian.net/browse/CHECKBOX-1497